networkx/algorithms/tree/tests/test_branchings.py

```diff
@@ -334,3 +334,46 @@ def test_edmonds1_minbranch():
     x = branchings.minimum_branching(G)
     x_ = build_branching(edges)
     assert_equal_branchings(x, x_)
+
+
+def test_edge_attribute_preservation():
+    # Test that edge attributes are preserved when finding an optimum graph
+    # using the Edmonds class. Check this for both normal and multigraphs
+    G = nx.Graph()
+
+    edgelist = [(0, 1, [('weight', 5), ('otherattr', 1), ('otherattr2', 3)]),
+                (0, 2, [('weight', 5), ('otherattr', 2), ('otherattr2', 2)]),
+                (1, 2, [('weight', 6), ('otherattr', 3), ('otherattr2', 1)])]
+    G.add_edges_from(edgelist)
+
+    ed = branchings.Edmonds(G)
+    B = ed.find_optimum('weight', preserve_attrs=True)
+
+    assert_equal(B[0][1]['otherattr'], 1)
+    assert_equal(B[0][1]['otherattr2'], 3)
+
+    G = nx.MultiGraph()
+    G.add_edges_from(edgelist * 2)  # Make sure we have duplicate edge paths
+
+    ed = branchings.Edmonds(G)
+    B = ed.find_optimum('weight', preserve_attrs=True)
+
+    assert_equal(B[0][1][0]['otherattr'], 1)
+    assert_equal(B[0][1][0]['otherattr2'], 3)
```

Comment:
Make this two tests?  Then, a failure for one but success for the other is more informative in the test output.

Reply:
Sounds sensibe